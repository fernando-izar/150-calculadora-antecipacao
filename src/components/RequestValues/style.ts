import styled from "styled-components";

export const Container = styled.div`    


    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: 20px;
    width: 60%;
    background-color: #ffffff;
`