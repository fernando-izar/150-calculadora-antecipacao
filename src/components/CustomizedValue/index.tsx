import { Container } from "./style";

export const CustomizedValue = () => {
  return (
    <Container>
      <h1>Customized Value</h1>
    </Container>
  );
}