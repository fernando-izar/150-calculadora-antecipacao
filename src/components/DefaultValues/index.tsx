import { Container } from "./style";

export const DefaultValues = () => {
  return (
    <Container>
      <h1>Default Values</h1>
    </Container>
  );
}