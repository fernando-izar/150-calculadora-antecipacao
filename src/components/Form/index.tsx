import {useContext} from "react";
import {useForm} from "react-hook-form"
import {yupResolver} from "@hookform/resolvers/yup";
import {schema} from "../../validators/schemas";
import {DefaultRequest} from "../../interfaces/requests";
import {MainContext} from "../../contexts/MainContext";
import { Container } from "./style";


export const Form = () => {
  const {onSubmitRequest} = useContext(MainContext);
  const {register, handleSubmit, formState: {errors}} = useForm<DefaultRequest>({
    resolver: yupResolver(schema)
  });

  return (
    <Container>
      <form onSubmit={handleSubmit(onSubmitRequest)}>
        <p>Informe o valor da venda *</p>
        <input className="input" placeholder="Valor da venda" type="number" {...register("amount")} />
        <p>{errors.amount?.message}</p>
        <p>Em quantas parcelas *</p>
        <input className="input" placeholder="Quantidade de parcelas"  type="number" {...register("installments")} />
        <p>{errors.installments?.message}</p>
        <p>Informe o percentual de MDR *</p>
        <input className="input" placeholder="Percentual de MDR" type="number" {...register("mdr")} />
        <p>{errors.mdr?.message}</p>
        <button className="form-button">Submit</button>
      </form>
    </Container>
  );
};