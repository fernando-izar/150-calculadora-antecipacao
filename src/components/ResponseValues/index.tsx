import { Container } from "./style";

export const ResponseValues = () => {
  return (
    <Container>
      <h1>Response Values</h1>
    </Container>
  );
}