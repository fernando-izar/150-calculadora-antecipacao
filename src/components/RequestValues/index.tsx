import { Form } from "../Form";
import { Container } from "./style";

export const RequestValues = () => {
  return (
    <Container>
      <h1>Simule sua Antecipação</h1>
      <Form/>
    </Container>
  );
}