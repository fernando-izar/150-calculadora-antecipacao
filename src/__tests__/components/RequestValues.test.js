import React from "react";
import renderer from "react-test-renderer";
import { RequestValues } from "../../components/RequestValues";

describe("RequestValues", () => {
    it("should not have unexpected changes", () => {
        const form = renderer.create(<RequestValues />).toJSON();
        expect(form).toMatchSnapshot();
    });
});