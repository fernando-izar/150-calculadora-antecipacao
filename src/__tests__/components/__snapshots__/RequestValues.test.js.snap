// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestValues should not have unexpected changes 1`] = `
<div
  className="sc-gswNZR ejEuEy"
>
  <h1>
    Simule sua Antecipação
  </h1>
  <div
    className="sc-bcXHqe Expvn"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          className="input-label"
          htmlFor="input-amount"
        >
          Informe o valor da venda *
        </label>
        <input
          className="input"
          id="input-amount"
          name="amount"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <p
          className="errors"
        />
      </div>
      <div>
        <label
          className="input-label"
          htmlFor="input-installments"
        >
          Em quantas parcelas *
        </label>
        <input
          className="input"
          id="input-installments"
          name="installments"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <p
          className="max-installments"
        >
          Máximo de 12 parcelas
        </p>
        <p
          className="errors"
        />
      </div>
      <div>
        <label
          className="input-label"
          htmlFor="input-mdr"
        >
          Informe o percentual de MDR *
        </label>
        <input
          className="input"
          id="input-mdr"
          name="mdr"
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
        />
        <p
          className="errors"
        />
      </div>
      <button
        className="form-button"
      >
        Calcular
      </button>
    </form>
  </div>
</div>
`;
